src/CodeGen_X86.cpp

```diff
@@ -94,10 +94,12 @@ void CodeGen_X86::visit(const GT *op) {
         // split it up ourselves.
 
         Type t = op->a.type();
-        int slice_size = 128 / t.bits();
-        if (slice_size < t.lanes()) {
-            slice_size = target.natural_vector_size(t);
-        }
+        int vec_bits = t.lanes() * t.bits();
```

Comment:
surely there's a `get_slice_size()` function waiting to be created here

Reply:
done